File: 4038-count-integers-appearing-in-a-single-block/4038-count-integers-appearing-in-a-single-block.java
class Solution {
    public int countSpecialIntegers(int[] nums) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int cnt=0;
        int prev=nums[0],s=0;
        for(int i:nums)
        mp.put(i,mp.getOrDefault(i,0)+1);
        for(int i:nums)
        {
            if(i!=prev)
            {
                s+=mp.get(prev)==cnt?1:0;
                prev=i;
                cnt=1;
            }
            else
            cnt++;
        }
        return cnt==mp.get(prev)?s+1:s;
    }
}

File: 4038-count-integers-appearing-in-a-single-block/4038-count-integers-appearing-in-a-single-block.cpp
class Solution {
public:
    int countSpecialIntegers(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(int i:nums)mp[i]++;
        int prev=nums[0];
        int cnt=0;
        int s=0;
        for(int i:nums)
        {
            if(i!=prev)
            {
                s+=cnt==mp[prev]?1:0;
                prev=i;
                cnt=1;
            }
            else
            cnt++;
        }
        return cnt==mp[prev]?s+1:s;
    }
};